src/main/java/com/vinylteam/vinyl/service/ConfirmationService.java

```diff
@@ -0,0 +1,21 @@
+package com.vinylteam.vinyl.service;
+
+import com.vinylteam.vinyl.entity.ConfirmationToken;
+
+import java.util.Optional;
+
+public interface ConfirmationService {
+
+    Optional<ConfirmationToken> findByToken(String token);
+
+    Optional<ConfirmationToken> findByUserId(long userId);
+
+    ConfirmationToken addByUserId(long userId);
```

Comment:
We should return optional.

Reply:
> We should return optional.

Why? DefaultConfirmationService, 
Row 55 ConfirmationToken newConfirmationToken = new ConfirmationToken(); 
We always return non-null object. Optional uses for avoid nullPointerException